### ISSUE TITLE
The root on ZFS howto for Debian does not work.

### ISSUE THREAD

**Issue by ellisonj** [zfsonlinux/zfs#8882]
@rlaager 

This is the live cd image running on a USB drive. 

Linux debian 4.9.0-9-amd64 #1 SMP Debian 4.9.168-1 (2019-04-12) x86_64 GNU/Linux
zfs-dkms/stretch-backports,now 0.7.12-1\~bpo9+1 all
spl-dkms/stretch-backports,now 0.7.12-1\~bpo9+1 all

### Describe the problem you're observing

Howto progresses with no issue until you get to: apt install --yes grub-pc 

It will not install to the raw device. 

Then you execute: grub-probe /boot after one intermediary step, and it fails. With "unknown filesystem." 

Three more steps you execute:  update-grub
Generating grub configuration file ...
Found linux image: /boot/vmlinuz-4.9.0-9-amd64
Found initrd image: /boot/initrd.img-4.9.0-9-amd64
/usr/sbin/grub-probe: error: unknown filesystem.
done

Same issue. 

#ls /boot/grub/*/zfs.mod
/boot/grub/i386-pc/zfs.mod

This would imply everything should work? I'm giving up on this now. I did this twice on two systems, and it fails in the same way. I hope you can fix this someday? I'll try again when I get some more free time.

**Comment by rlaager** [comment]
Please paste the output of: `zpool get all rpool | grep feature`

**Comment by rlaager** [comment]
Sorry, that should be the bpool: `zpool get all rpool | grep feature`

But just paste both please.

**Comment by rlaager** [comment]
Let me try again. Please paste the output of: `zpool get all | grep feature`

**Comment by ellisonj** [comment]
I already wiped the disks. I used your exact howto except "-o ashift=12" since they were not 4k disks. They were SSDs.

**Comment by rlaager** [comment]
When you get a chance to test again, let me know. You might want to jump into #zfsonlinux on Freenode for live(-ish) discussion.

That error is almost certainly occurring because `/boot` is on a pool with a feature not supported by GRUB. I’d like to rule that in or out first.

**Comment by kpande** [comment]
use `zpool create -d` for the bpool to disable all feature flags to ensure it will load in grub.

**Comment by ellisonj** [comment]
If that is the case then that is what you must fix on your guide. I literally copy pasted every command I used. 

 359  zpool create -d       -o feature@async_destroy=enabled       -o feature@bookmarks=enabled       -o feature@embedded_data=enabled       -o feature@empty_bpobj=enabled       -o feature@enabled_txg=enabled       -o feature@extensible_dataset=enabled       -o feature@filesystem_limits=enabled       -o feature@hole_birth=enabled       -o feature@large_blocks=enabled       -o feature@lz4_compress=enabled       -o feature@spacemap_histogram=enabled -o feature@userobj_accounting=enabled       -O acltype=posixacl -O canmount=off -O compression=lz4 -O devices=off       -O normalization=formD -O relatime=on -O xattr=sa       -O mountpoint=/ -R /mnt       bpool mirror /dev/sdb3 /dev/sdc3
  

  362  zpool create -O acltype=posixacl -O canmount=off -O compression=lz4 -O dnodesize=auto -O normalization=formD -O relatime=on -O xattr=sa       -O mountpoint=/ -R /mnt       rpool mirror /dev/sdb4 /dev/sdc4

**Comment by ellisonj** [comment]
How about someone use the latest live for Debian on a BIOS server, and see how that goes for you? Use mirrored SSDs.

**Comment by rlaager** [comment]
> If that is the case then that is what you must fix on your guide. I literally copy pasted every command I used.
> 
> 359 zpool create -d -o feature@async_destroy=enabled -o feature@bookmarks=enabled -o feature@embedded_data=enabled -o feature@empty_bpobj=enabled -o feature@enabled_txg=enabled -o feature@extensible_dataset=enabled -o feature@filesystem_limits=enabled -o feature@hole_birth=enabled -o feature@large_blocks=enabled -o feature@lz4_compress=enabled -o feature@spacemap_histogram=enabled -o feature@userobj_accounting=enabled -O acltype=posixacl -O canmount=off -O compression=lz4 -O devices=off -O normalization=formD -O relatime=on -O xattr=sa -O mountpoint=/ -R /mnt bpool mirror /dev/sdb3 /dev/sdc3

That looks right to me. You have `-d` in there to start with no features, and you're only enabling features that GRUB supports reading. That matches the guide (minus `-o ashift=12` as mentioned already, which isn't at issue here).

I upgraded my stretch test VM, rebooted, and checked with `grub-probe /boot`. It's still working. There could still be some subtle difference in the live CD, of course.

**Comment by ellisonj** [comment]
Just confirmed the older ISO is not available anymore. Sigh.... I always seem to find time for these things right after they stop working.

**Comment by rlaager** [comment]
For anyone driving by, we worked on this via email. Ultimately it turned out to be a corrupt installer image due to a failing USB flash drive. It was really bizarre, but things are working now!
